src/Xamarin.Android.Tools.AndroidSdk/Sdks/AndroidSdkUnix.cs

```diff
@@ -178,22 +179,30 @@ public override void SetPreferredAndroidNdkPath (string path)
 			}
 
 			androidEl.SetAttributeValue ("path", path);
-			SaveConfig (doc);
+			SaveConfig (doc, Logger);
 		}
 
-		void SaveConfig (XDocument doc)
+		bool SaveConfig (XDocument doc, Action<TraceLevel, string> logger)
 		{
 			string cfg = UnixConfigPath;
 			List <string> created = null;
 
 			if (!File.Exists (cfg)) {
 				string dir = Path.GetDirectoryName (cfg);
 				if (!Directory.Exists (dir)) {
-					Directory.CreateDirectory (dir);
-					AddToList (dir);
-				}
-				AddToList (cfg);
-			}
+                                try
+                                {
+                                    Directory.CreateDirectory(dir);
+                                }
+                                catch (Exception ex)
+                                {
+                                    logger(TraceLevel.Error, $"Unable to create directory {dir}, user do not have the right permissions, {ex.Message}.");
+                                    return false;
+                                }
+                                    AddToList(dir);
+                        }   
+        	   	AddToList (cfg);
```

Comment:
This file uses *tabs*, not spaces, for indentation.  All the indentation and associated formatting is wrong.